core-services/src/main/java/org/zalando/nakadi/service/publishing/EventPublisher.java

```diff
@@ -133,8 +135,7 @@ EventPublishResult processInternal(final String events,
             if (!delete) {
                 enrich(batch, eventType);
             }
-            eventOwnerExtractor.extract(batch, eventType);
-            authorizeEventWrite(batch);
+            validateEventOwnership(eventType, batch);
```

Comment:
the validation happens before partitioning and enrichment, it would make more sense to move `validateEventOwnership` close to `validate`